google/cloud/forseti/scanner/audit/kms_rules_engine.py

```diff
@@ -382,30 +447,61 @@ def find_violations(self, key):
             list: Returns a list of RuleViolation named tuples.
         """
         violations = []
-        creation_time = key.primary_version.get('createTime')
         state = key.primary_version.get('state')
-        if not state == 'ENABLED':
-            return violations
-        scan_time = date_time.get_utc_now_datetime()
-        if self.rule['mode'] == BLACKLIST:
-            if not self.is_key_rotated(creation_time, scan_time):
-                violation_reason = ('Key %s was not rotated since %s.' %
-                                    (key.name, creation_time))
-                violations.append(RuleViolation(
-                    resource_id=key.id,
-                    resource_type=key.type,
-                    resource_name=key.id,
-                    full_name=key.crypto_key_full_name,
-                    rule_index=self.rule_index,
-                    rule_name=self.rule_name,
-                    violation_type=VIOLATION_TYPE,
-                    primary_version=key.primary_version,
-                    next_rotation_time=key.next_rotation_time,
-                    rotation_period=key.rotation_period,
-                    violation_reason=violation_reason,
-                    key_creation_time=key.create_time,
-                    version_creation_time=creation_time,
-                    resource_data=key.data))
+
+        mode = self.rule['mode']
+        has_violation = False
+
+        crypto_key = self.rule['key']
```

Comment:
crypto_key can be confusing here, crypto_key_rule might be better

Reply:
Ah, ok. Renamed.